tests/FakeItEasy.Specs/CreationSpecs.cs

```diff
@@ -177,5 +204,10 @@ protected override IList<T> CreateCollectionOfFake<T>(int numberOfFakes)
         {
             return A.CollectionOfFake(typeof(T), numberOfFakes).Cast<T>().ToList();
         }
+
+        protected override IList<T> CreateCollectionOfFake<T>(int numberOfFakes, Action<IFakeOptions<T>> optionsBuilder)
+        {
+            return A.CollectionOfFake(typeof(T), numberOfFakes, options => optionsBuilder((IFakeOptions<T>)options)).Cast<T>().ToList();

```

Comment:
[This code](https://gist.github.com/thomaslevesque/a4ad1fc87dd03edfb3b8dfc589146156) works, but if you remove the inheritance between the interfaces, it doesn't.